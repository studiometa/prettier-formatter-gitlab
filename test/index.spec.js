import { describe, it, expect, afterEach, spyOn } from 'bun:test';
import { existsSync, unlinkSync } from 'node:fs';
import { prettierFormatterGitLab } from '../src/index.js';

const CODE_QUALITY_FILENAME = 'gl-prettier-codequality.json';

afterEach(() => {
  if (existsSync(CODE_QUALITY_FILENAME)) {
    unlinkSync(CODE_QUALITY_FILENAME);
  }
});

describe('The prettierFormatterGitLab function', () => {
  it('should work with the `-c` or `-l` flag', async () => {
    const consoleSpy = spyOn(console, 'log');
    consoleSpy.mockImplementation(() => null);
    process.env.PRETTIER_CODE_QUALITY_REPORT = CODE_QUALITY_FILENAME;

    // prettier -c path/to/folder/
    const outputCheck = `Checking formatting...
[warn] test/__stubs__/dirty-2.js
[warn] test/__stubs__/dirty.js
test/__stubs__/syntax-error.js
[error] test/__stubs__/syntax-error.js: SyntaxError: Unexpected token, expected "(" (3:13)
[error]   1 | /* eslint-disable */
[error]   2 |
[error] > 3 | function foo#####( arg  ) {
[error]     |             ^
[error]   4 |   return arg;
[error]   5 | }
[error]   6 |
[warn] Code style issues found in 2 files. Run Prettier to fix.
`;

    // prettier -l path/to/folder/
    const outputList = `test/__stubs__/dirty-2.js
test/__stubs__/dirty.js
test/__stubs__/syntax-error.js
`;

    await prettierFormatterGitLab(outputCheck);
    await prettierFormatterGitLab(outputList);
    expect(consoleSpy.mock.calls).toMatchSnapshot();
    consoleSpy.mockRestore();
  });
});
