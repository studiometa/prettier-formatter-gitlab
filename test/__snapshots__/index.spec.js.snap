// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`The prettierFormatterGitLab function should work with the \`-c\` or \`-l\` flag 1`] = `
[
  [
    
"----------------------------------------------------
test/__stubs__/dirty-2.js
----------------------------------------------------
@@ -1,7 +1,7 @@
  /* eslint-disable */

- function foo( arg  ) {
+ function foo(arg) {
    const data = arg.split(',');
-   const message = \`Cras non justo iaculis, placerat tortor non, venenatis neque. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Nullam ut neque a justo efficitur tristique. Aliquam consequat convallis sagittis. Curabitur venenatis imperdiet fermentum. Curabitur euismod sit amet mi eu tristique. Phasellus leo nisi, tempor efficitur laoreet vitae, rutrum id justo. Nullam ac mi vehicula, rhoncus turpis a, mattis odio. Donec id condimentum metus. Vivamus posuere libero nec orci volutpat, in euismod dolor ornare. Cras tincidunt gravida ante, sed dictum nulla posuere ut.\`
+   const message = \`Cras non justo iaculis, placerat tortor non, venenatis neque. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Nullam ut neque a justo efficitur tristique. Aliquam consequat convallis sagittis. Curabitur venenatis imperdiet fermentum. Curabitur euismod sit amet mi eu tristique. Phasellus leo nisi, tempor efficitur laoreet vitae, rutrum id justo. Nullam ac mi vehicula, rhoncus turpis a, mattis odio. Donec id condimentum metus. Vivamus posuere libero nec orci volutpat, in euismod dolor ornare. Cras tincidunt gravida ante, sed dictum nulla posuere ut.\`;
    return data;
  }
"
,
  ],
  [
    
"----------------------------------------------------
test/__stubs__/dirty.js
----------------------------------------------------
@@ -1,18 +1,7 @@
  /* eslint-disable */

-
-
-
-
-
-
- function foo( arg  ) {
+ function foo(arg) {
-
-
    const data = arg.split(',');

-
-
-
    return data;
  }
"
,
  ],
  [
    
"----------------------------------------------------
test/__stubs__/syntax-error.js
----------------------------------------------------
SyntaxError: Unexpected token, expected "(" (3:13)
\x1B[0m \x1B[90m 1 |\x1B[39m \x1B[90m/* eslint-disable */\x1B[39m
 \x1B[90m 2 |\x1B[39m
\x1B[31m\x1B[1m>\x1B[22m\x1B[39m\x1B[90m 3 |\x1B[39m \x1B[36mfunction\x1B[39m foo\x1B[33m#\x1B[39m\x1B[33m#\x1B[39m\x1B[33m#\x1B[39m\x1B[33m#\x1B[39m\x1B[33m#\x1B[39m( arg  ) {
 \x1B[90m   |\x1B[39m             \x1B[31m\x1B[1m^\x1B[22m\x1B[39m
 \x1B[90m 4 |\x1B[39m   \x1B[36mreturn\x1B[39m arg\x1B[33m;\x1B[39m
 \x1B[90m 5 |\x1B[39m }
 \x1B[90m 6 |\x1B[39m\x1B[0m
"
,
  ],
  [
    
"----------------------------------------------------
test/__stubs__/syntax-error.js
----------------------------------------------------
SyntaxError: Unexpected token, expected "(" (3:13)
\x1B[0m \x1B[90m 1 |\x1B[39m \x1B[90m/* eslint-disable */\x1B[39m
 \x1B[90m 2 |\x1B[39m
\x1B[31m\x1B[1m>\x1B[22m\x1B[39m\x1B[90m 3 |\x1B[39m \x1B[36mfunction\x1B[39m foo\x1B[33m#\x1B[39m\x1B[33m#\x1B[39m\x1B[33m#\x1B[39m\x1B[33m#\x1B[39m\x1B[33m#\x1B[39m( arg  ) {
 \x1B[90m   |\x1B[39m             \x1B[31m\x1B[1m^\x1B[22m\x1B[39m
 \x1B[90m 4 |\x1B[39m   \x1B[36mreturn\x1B[39m arg\x1B[33m;\x1B[39m
 \x1B[90m 5 |\x1B[39m }
 \x1B[90m 6 |\x1B[39m\x1B[0m
"
,
  ],
  [
    
"----------------------------------------------------
test/__stubs__/dirty-2.js
----------------------------------------------------
@@ -1,7 +1,7 @@
  /* eslint-disable */

- function foo( arg  ) {
+ function foo(arg) {
    const data = arg.split(',');
-   const message = \`Cras non justo iaculis, placerat tortor non, venenatis neque. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Nullam ut neque a justo efficitur tristique. Aliquam consequat convallis sagittis. Curabitur venenatis imperdiet fermentum. Curabitur euismod sit amet mi eu tristique. Phasellus leo nisi, tempor efficitur laoreet vitae, rutrum id justo. Nullam ac mi vehicula, rhoncus turpis a, mattis odio. Donec id condimentum metus. Vivamus posuere libero nec orci volutpat, in euismod dolor ornare. Cras tincidunt gravida ante, sed dictum nulla posuere ut.\`
+   const message = \`Cras non justo iaculis, placerat tortor non, venenatis neque. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Nullam ut neque a justo efficitur tristique. Aliquam consequat convallis sagittis. Curabitur venenatis imperdiet fermentum. Curabitur euismod sit amet mi eu tristique. Phasellus leo nisi, tempor efficitur laoreet vitae, rutrum id justo. Nullam ac mi vehicula, rhoncus turpis a, mattis odio. Donec id condimentum metus. Vivamus posuere libero nec orci volutpat, in euismod dolor ornare. Cras tincidunt gravida ante, sed dictum nulla posuere ut.\`;
    return data;
  }
"
,
  ],
  [
    
"----------------------------------------------------
test/__stubs__/dirty.js
----------------------------------------------------
@@ -1,18 +1,7 @@
  /* eslint-disable */

-
-
-
-
-
-
- function foo( arg  ) {
+ function foo(arg) {
-
-
    const data = arg.split(',');

-
-
-
    return data;
  }
"
,
  ],
  [
    
"----------------------------------------------------
test/__stubs__/syntax-error.js
----------------------------------------------------
SyntaxError: Unexpected token, expected "(" (3:13)
\x1B[0m \x1B[90m 1 |\x1B[39m \x1B[90m/* eslint-disable */\x1B[39m
 \x1B[90m 2 |\x1B[39m
\x1B[31m\x1B[1m>\x1B[22m\x1B[39m\x1B[90m 3 |\x1B[39m \x1B[36mfunction\x1B[39m foo\x1B[33m#\x1B[39m\x1B[33m#\x1B[39m\x1B[33m#\x1B[39m\x1B[33m#\x1B[39m\x1B[33m#\x1B[39m( arg  ) {
 \x1B[90m   |\x1B[39m             \x1B[31m\x1B[1m^\x1B[22m\x1B[39m
 \x1B[90m 4 |\x1B[39m   \x1B[36mreturn\x1B[39m arg\x1B[33m;\x1B[39m
 \x1B[90m 5 |\x1B[39m }
 \x1B[90m 6 |\x1B[39m\x1B[0m
"
,
  ],
]
`;
